src/RA_Core.cpp

```diff
@@ -1545,37 +1548,24 @@ void _WriteBufferToFile(const std::string& sFileName, const Document& doc)
     }
 }
 
-void _WriteBufferToFile(const std::string& sFileName, const DataStream& raw)
+void _WriteBufferToFile(const std::string& sFileName, const std::string& raw)
 {
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, sFileName.c_str(), "wb") == 0)
-    {
-        fwrite(raw.data(), 1, raw.size(), pf);
-        fclose(pf);
-    }
-}
 
-void _WriteBufferToFile(const std::string& sFileName, const std::string& sData)
-{
-    SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, sFileName.c_str(), "wb") == 0)
-    {
-        fwrite(sData.data(), 1, sData.length(), pf);
-        fclose(pf);
-    }
+    using FileH = std::unique_ptr<FILE, decltype(&std::fclose)>;
+    FileH myFile{ std::fopen(sFileName.c_str(), "wb"), std::fclose };
+    std::fwrite(static_cast<const void*>(raw.c_str()), sizeof(char), raw.length(), myFile.get());
 }
 
-void _WriteBufferToFile(const char* sFile, const BYTE* sBuffer, int nBytes)
+
+void _WriteBufferToFile(const char* sFile, std::streamsize nBytes)
 {
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, sFile, "wb") == 0)
-    {
-        fwrite(sBuffer, 1, nBytes, pf);
-        fclose(pf);
-    }
+
+    using FileH = std::unique_ptr<FILE, decltype(&std::fclose)>;
+    FileH myFile{ std::fopen(sFile, "wb"), std::fclose };
+    auto sBuffer{ std::make_unique<char[]>(ra::to_unsigned(nBytes)) };
+    std::fwrite(static_cast<void* const>(sBuffer.get()), sizeof(char), std::strlen(sBuffer.get()), myFile.get());
```

Comment:
Wasn't sure if I should've removed this, even with original signature it was referenced anywhere.

Reply:
It looks like it was intended to be called from a couple places in RA_Core.cpp that are currently commented out. I'm fine with removing it - less code to test. We have git history if we ever need to add it back.